technologies/job/gcp-cloud-functions/project/instanceActions.js

```diff
@@ -0,0 +1,81 @@
+const { Response, JobStatus, Log } = require('@saagie/sdk');
+const { google } = require('googleapis');
+const cloudfunctions = google.cloudfunctions('v1');
+const logging = google.logging('v2');
+const { getConnexion } = require('./utils');
+
+
+/**
+ * Logic to start the external job instance.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data.
+ */
+exports.start = async ({ job, instance }) => {
+  try {
+    // You can return any payload you want to get in the stop and getStatus functions.
+    return Response.success({ customId: "" });
+  } catch (error) {
+    return Response.error('Fail to start job', { error });
+  }
+};
+
+/**
+ * Logic to retrieve the external job instance status.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data including the payload returned in the start function.
+ */
+exports.getStatus = async ({ job }) => {
+  try {
+    const gcpKey = JSON.parse(job.featuresValues.endpoint.jsonKey);
+
+    authClient = getConnexion(gcpKey);
+
+    console.log(`Retrieve status for ${job.featuresValues.function.id}`)
+    const { data } = await cloudfunctions.projects.locations.functions.get({
+      auth: authClient,
+      name : `${job.featuresValues.function.id}`,
+    });
+
+    switch (data.status) {
+      case 'ACTIVE':
+        return Response.success(JobStatus.RUNNING);
+      case 'OFFLINE':
+        return Response.success(JobStatus.FAILED);
+      case 'DEPLOY_IN_PROGRESS':
+        return Response.success(JobStatus.AWAITING);
+      default:
+        return Response.success(JobStatus.KILLED);
+    }
```

Comment:
```suggestion
    const JOB_STATUS = {
        'ACTIVE': JobStatus.RUNNING,
        'OFFLINE': JobStatus.FAILED,
        'DEPLOY_IN_PROGRESS': JobStatus.AWAITING,
    };

    return Response.success(JOB_STATUS[data.status] || JobStatus.KILLED);
```

Useful to directly see which Saagie status correspond to which Google Cloud Functions status.
Using `||`, if `JOB_STATUS[data.status]` does not exists, then `JobStatus.KILLED` will be the default.